tests/jobs/misfit_preprocessor/unit/test_config.py

```diff
@@ -327,3 +328,21 @@ def test_config_workflow_valid(workflow):
     config_data = {"workflow": {"type": workflow}}
     config = workflow_config.MisfitConfig(**config_data)
     assert config.workflow.type == workflow
+
+
+@pytest.mark.parametrize("workflow", ["auto_scale", "custom_scale"])
+@pytest.mark.parametrize(
+    "clustering_config",
+    [
+        {},
+        {"init": "random"},
+        {"init": "k-means++"},
+        {"n_init": 1},
+        {"max_iter": 10},
+        {"random_state": 10},
+    ],
+)
+def test_kmeans_schema(workflow, clustering_config):
```

Comment:
Just testing that what you set is what you get? And that they make valid configs? Just to get a bit more out of the test, you could verify that the rest are kept as defaults. Not a game changer, but simply a suggestion.

Could also test some invalid configs, for instance several of these are specified to be constrained types, verify that the constraints are correct?

Reply:
I can add a couple of other tests for what you suggest, I would like to keep this test as is to avoid testing too many things at once.